src/goto-instrument/contracts/assigns.cpp

```diff
@@ -52,6 +71,35 @@ static const slicet normalize_to_slice(const exprt &expr, const namespacet &ns)
   UNREACHABLE;
 }
 
+/// Normalises a assigns target expression to a guarded slice struct.
+///
+/// ```
+/// case expr of
+///  | guard ? target ->
+///          {guard
+///           target,
+///           normalize_to_slice(target)}
+///  | target ->
+///          {true,
+///           target,
+///           normalize_to_slice(target)}
+/// ```
+static const guarded_slicet
+normalize_to_guarded_slice(const exprt &expr, const namespacet &ns)
+{
+  if(expr.id() == ID_conditional_assigns_target)
+  {
+    const auto &new_expr = expr.operands().at(1);
+    const auto slice = normalize_to_slice(new_expr, ns);
+    return {expr.operands().at(0), new_expr, slice.first, slice.second};
+  }
+  else
+  {
+    const auto slice = normalize_to_slice(expr, ns);
+    return {true_exprt{}, expr, slice.first, slice.second};
+  }
+}
```

Comment:
I suppose we could get rid of this if the the front-end always generated conditional assigns targets?